### PR TITLE
add trace id

### PR DESCRIPTION
**Pull request by lvjing2**
add trace id for reconcile, cause the knative/key is not enough yet if it reconcile for the same key for many times.

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -277,7 +278,7 @@ func (c *Impl) processNextWorkItem() bool {
 
 	// Embed the key into the logger and attach that to the context we pass
 	// to the Reconciler.
-	logger := c.logger.With(zap.String(logkey.Key, key))
+	logger := c.logger.With(zap.String(logkey.TraceId, uuid.New().String())).With(zap.String(logkey.Key, key))
```
Comment (vagababov) [inline controller/controller.go]: `With` takes a list.

Comment (lvjing2) [reply controller/controller.go]: oh, nice tip

Comment (lvjing2) [reply controller/controller.go]: done

File: logging/logkey/constants.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ const (
 	// Key is the key (namespace/name) being reconciled.
 	Key = "knative.dev/key"
 
+	TraceId = "knative.dev/traceid"
```
Comment (mdemirhan) [inline logging/logkey/constants.go]: Can you please add the documentation for this? Something like:

// TraceId is used to track an asynchronous or long running operation within Knative.

Comment (lvjing2) [reply logging/logkey/constants.go]: ok, done

**Pull request merged**